src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.component.html

```diff
@@ -0,0 +1,72 @@
+<h1>
+    <!--Default IDE Settings-->
+    {{ 'artemisApp.userSettings.idePreferences' | artemisTranslate }}
+</h1>
+<div class="list-group d-block">
+    <div class="list-group-item pb-3">
+        <dt>
+            <span jhiTranslate="artemisApp.userSettings.idePreferencesPage.description"> </span>
+        </dt>
+        <dt class="py-2">
+            {{ 'artemisApp.userSettings.idePreferencesPage.default' | artemisTranslate }}
+        </dt>
+        <div class="btn-group">
+            @for (ide of PREDEFINED_IDE; track ide) {
+                <div
+                    class="btn btn-default"
+                    [ngClass]="{
+                        'btn-primary selected': isIdeOfProgrammingLanguage(ProgrammingLanguage.EMPTY, ide),
+                        'btn-default': !isIdeOfProgrammingLanguage(ProgrammingLanguage.EMPTY, ide),
+                    }"
+                    (click)="changeIde(ProgrammingLanguage.EMPTY, ide)"
+                >
+                    {{ ide.name }}
+                </div>
+            }
+        </div>
+    </div>
+    <!-- Display each customized Programming Language -->
+    @for (programmingLanguage of getCustomizedProgrammingLanguages(); track programmingLanguage) {
+        <div class="list-group-item pb-3">
+            <div class="item-container d-flex flex-row justify-content-between align-items-center pb-2">
+                <dt>
+                    {{ programmingLanguage }}
+                </dt>
+                <button class="btn btn-sm btn-outline-danger" (click)="removeProgrammingLanguage(programmingLanguage)">
+                    <fa-icon [icon]="faTrash"></fa-icon>
+                </button>
+            </div>
+            <div class="btn-group">
+                @for (ide of PREDEFINED_IDE; track ide) {
+                    <div
+                        class="btn btn-default"
+                        [ngClass]="{
+                            'btn-primary selected': isIdeOfProgrammingLanguage(programmingLanguage, ide),
+                            'btn-default': !isIdeOfProgrammingLanguage(programmingLanguage, ide),
+                        }"
+                        (click)="changeIde(programmingLanguage, ide)"
+                    >
+                        {{ ide.name }}
+                    </div>
+                }
+            </div>
+        </div>
+    }
+    <!-- Display plus button to add Programming Language with drop down menu -->
+    @if (getRemainingProgrammingLanguages().length != 0) {
+        <div class="list-group-item">
+            <div ngbDropdown class="d-inline-block">
+                <button type="button" class="btn btn-outline-primary" id="dropdown" ngbDropdownToggle>
+                    <fa-icon [icon]="faPlus"></fa-icon>
+                </button>
+                <div ngbDropdownMenu aria-labelledby="dropdown">
+                    @for (programmingLanguage of getRemainingProgrammingLanguages(); track programmingLanguage) {
```

Comment:
More method calls - could we avoid them?

Reply:
resolved